src/hdf5/HDF5IO.cpp

```diff
@@ -377,6 +379,21 @@ Status HDF5IO::createStringDataSet(const std::string& path,
   return Status::Success;
 }
 
+Status HDF5IO::startRecording()
+{
+  modifyObjects = false;
```

Comment:
Related to the above comment, I think in `HDF5IO`, instead of having to set the `modifyObjects` flag manually, we could implement `canModifyObjects` to check if `SWMR` mode is enabled. In this way we can avoid potential inconsistencies between the mode of the file and the status of `modifyObjects` flag. I.e., I think `canModifyObjects` would looks as follows:


```
bool HDF5IO::canModifyObjects()
{
   // Get the file access intent
    unsigned int intent;
    herr_t status = H5Fget_intent(file_id, &intent);
    if (status < 0) {
        return false;  // We could not access the file so modifying objects is not going to work
    }

    // Check if SWMR mode is enabled
    if (intent & H5F_ACC_SWMR_READ) {
        return False;  // File is in SWMR mode
    } else {
        return true;  // File is not in SWMR mode
    }
}
```

Reply:
That makes sense to avoid inconsistencies. I added a `canModifyObjects` function with a note for derived classes to override the `BaseIO` implementation if needed